code/modules/mob/living/simple_animal/hostile/illusion.dm

```diff
@@ -16,51 +16,47 @@
 	health = 100
 	speed = 0
 	faction = list(FACTION_ILLUSION)
-	var/life_span = INFINITY //how long until they despawn
-	var/mob/living/parent_mob
-	var/multiply_chance = 0 //if we multiply on hit
-	del_on_death = 1
+	del_on_death = TRUE
 	death_message = "vanishes into thin air! It was a fake!"
+	/// Weakref to what we're copying
+	var/datum/weakref/parent_mob_ref
+	/// Prob of getting a clone on attack
+	var/multiply_chance = 0
 
-
-/mob/living/simple_animal/hostile/illusion/Life(seconds_per_tick = SSMOBS_DT, times_fired)
-	..()
-	if(world.time > life_span)
-		death()
-
-
-/mob/living/simple_animal/hostile/illusion/proc/Copy_Parent(mob/living/original, life = 50, hp = 100, damage = 0, replicate = 0 )
+/mob/living/simple_animal/hostile/illusion/proc/Copy_Parent(mob/living/original, life = 5 SECONDS, hp = 100, damage = 0, replicate = 0)
```

Comment:
would you mind turning this into `snake_case` or do you think we should just wait until someone comes by and refactors this into a basic mob lol. ultimately optional but i figured i'd mention it given how wack it is

Reply:
I don't really want to update the consumers of this proc because it only causes me more pain on the swing branch

Reply:
that's fine then